[9.x] Add missing php extensions in composer

joelharkes opened the pull request:
Followup: https://github.com/laravel/framework/pull/45848

add missing necessary (for testing) and suggested PHP extensions.

joelharkes committed:
Add missing php extensions

driesvints commented on composer.json:
```diff
@@ -83,6 +83,8 @@
     },
     "require-dev": {
+        "ext-gmp": "*",
+        "ext-bcmath": "*",
```
Didn't we recently removed these? @timacdonald

timacdonald replied:
We were still referencing bcmath deps in some random tests. I've refactored those away now.

taylorotwell committed:
Update composer.json

taylorotwell commented:
Waiting on @timacdonald review.

timacdonald commented:
There is some additional tweaking I would like to do for this one. Will take a look on Monday. Thanks

timacdonald committed:
adds hash-ext to Auth

timacdonald committed:
adds hash-ext to Cookie

timacdonald committed:
Removes always on json-ext from Encryption

- https://wiki.php.net/rfc/always_enable_json
- https://www.php.net/ChangeLog-8.php

timacdonald committed:
adds hash-ext to Encryption

timacdonald committed:
Adds ext-hash to Filesystem

timacdonald committed:
Adds ext-fileinfo to Filesystem

timacdonald committed:
Sorts Filesystem extensions

timacdonald committed:
Removes always on json-ext from Mail

timacdonald committed:
Removes always on json-ext from Pagination

timacdonald committed:
Adds ext-filter to Pagination

timacdonald committed:
Removes always on json-ext from Queue

timacdonald committed:
Adds ext-PDO to Queue

timacdonald committed:
Adds ext-mbstring to Queue

timacdonald committed:
Adds ext-filter to Queue

timacdonald committed:
Removes always on ext-json from Routing

timacdonald committed:
Adds ext-filter to Routing

timacdonald committed:
Adds ext-hash to Routing

timacdonald committed:
Removes always on json-ext from  Support

timacdonald committed:
Adds ext-ctype to Support

timacdonald committed:
Adds ext-filter to Support

timacdonald committed:
Removes always on ext-json from Translation

timacdonald committed:
Removes always on ext-json from View

timacdonald committed:
Adds ext-tokenizer to View

timacdonald committed:
Removes always on json-ext from Broadcasting

timacdonald committed:
Adds ext-hash to Broadcasting

timacdonald committed:
Adds ext-filter to Cache

timacdonald committed:
Adds ext-mbstring to Console

timacdonald committed:
Adds ext-pcntl to Console

timacdonald committed:
Removes always on ext-json from Database

timacdonald committed:
Adds ext-PDO to Database

timacdonald committed:
Adds ext-filter to Database

timacdonald committed:
Adds ext-pcntl to Database

timacdonald committed:
Remove always on ext-json from Http

timacdonald committed:
Adds ext-filter to Http

timacdonald committed:
Removes always on ext-json from Session

timacdonald committed:
Adds ext-ctype to Session

timacdonald committed:
Adds ext-session to Session

timacdonald committed:
Adds ext-mbstring to Testing

timacdonald committed:
Removes always on ext-json from Validation

timacdonald committed:
Adds ext-filter to Validation

timacdonald committed:
Adds ext-mbstring to Validation

timacdonald commented on src/Illuminate/Auth/composer.json:
```diff
@@ -15,6 +15,7 @@
     ],
     "require": {
         "php": "^8.0.2",
+        "ext-hash": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Auth/TokenGuard.php#L85

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Auth/DatabaseUserProvider.php#L76

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Auth/EloquentUserProvider.php#L83

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Auth/Passwords/DatabaseTokenRepository.php#L214

timacdonald commented on src/Illuminate/Broadcasting/composer.json:
```diff
@@ -15,7 +15,7 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-hash": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Broadcasting/Broadcasters/PusherBroadcaster.php#L57

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php#L109

timacdonald commented on src/Illuminate/Cache/composer.json:
```diff
@@ -15,6 +15,7 @@
     ],
     "require": {
         "php": "^8.0.2",
+        "ext-filter": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Cache/DynamoDbStore.php#L486

timacdonald commented on src/Illuminate/Console/composer.json:
```diff
@@ -15,6 +15,8 @@
     ],
     "require": {
         "php": "^8.0.2",
+        "ext-mbstring": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/View/Components/Task.php#L27

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/View/Components/Task.php#L44

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Scheduling/ScheduleListCommand.php#L103

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Scheduling/ScheduleListCommand.php#L116

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Scheduling/ScheduleListCommand.php#L130-L132

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Scheduling/ScheduleListCommand.php#L151

timacdonald commented on src/Illuminate/Console/composer.json:
```diff
@@ -15,6 +15,8 @@
     "require": {
         "php": "^8.0.2",
+        "ext-mbstring": "*",
+        "ext-pcntl": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Signals.php#L79

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Signals.php#L95

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Signals.php#L79

timacdonald commented on src/Illuminate/Cookie/composer.json:
```diff
@@ -15,6 +15,7 @@
     ],
     "require": {
         "php": "^8.0.2",
+        "ext-hash": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Cookie/CookieValuePrefix.php#L16

timacdonald commented on src/Illuminate/Database/composer.json:
```diff
@@ -16,7 +16,9 @@
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-PDO": "*",
+        "ext-filter": "*",
```
https://github.com/laravel/framework/blob/9.x/src/Illuminate/Database/Query/Grammars/PostgresGrammar.php#L231

https://github.com/laravel/framework/blob/9.x/src/Illuminate/Database/Query/Grammars/PostgresGrammar.php#L673

timacdonald commented on src/Illuminate/Database/composer.json:
```diff
@@ -16,7 +16,9 @@
-        "ext-json": "*",
+        "ext-PDO": "*",
+        "ext-filter": "*",
+        "ext-pcntl": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Database/Console/DatabaseInspectionCommand.php#L241

timacdonald commented on src/Illuminate/Database/composer.json:
```diff
@@ -16,7 +16,9 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-PDO": "*",
```
Search usages of `\PDO`, `\PDOStatement`, and `\PDOException`.

timacdonald commented on src/Illuminate/Encryption/composer.json:
```diff
@@ -15,7 +15,7 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-hash": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Encryption/Encrypter.php#L257

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Encryption/Encrypter.php#L195

timacdonald commented on src/Illuminate/Filesystem/composer.json:
```diff
@@ -15,6 +15,8 @@
     ],
     "require": {
         "php": "^8.0.2",
+        "ext-fileinfo": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Filesystem/Filesystem.php#L452

timacdonald commented on src/Illuminate/Filesystem/composer.json:
```diff
@@ -15,6 +15,8 @@
     "require": {
         "php": "^8.0.2",
+        "ext-fileinfo": "*",
+        "ext-hash": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Filesystem/Filesystem.php#L177

timacdonald commented on src/Illuminate/Http/composer.json:
```diff
@@ -15,7 +15,7 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-filter": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Http/Concerns/InteractsWithInput.php#L341-L344

timacdonald commented on src/Illuminate/Pagination/composer.json:
```diff
@@ -15,7 +15,7 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-filter": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Pagination/AbstractPaginator.php#L131-L134

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Pagination/PaginationState.php#L19-L27

timacdonald commented on src/Illuminate/Queue/composer.json:
```diff
@@ -15,7 +15,9 @@
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-PDO": "*",
+        "ext-filter": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/SqsQueue.php#L205-L212

timacdonald commented on src/Illuminate/Queue/composer.json:
```diff
@@ -15,7 +15,9 @@
-        "ext-json": "*",
+        "ext-PDO": "*",
+        "ext-filter": "*",
+        "ext-mbstring": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/Failed/DatabaseFailedJobProvider.php#L56-L65

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/Failed/DatabaseUuidFailedJobProvider.php#L56-L68

timacdonald commented on src/Illuminate/Queue/composer.json:
```diff
@@ -15,7 +15,9 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-PDO": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/DatabaseQueue.php#L256-L280

X-Coder264 replied:
We shouldn't require the PDO extension for the entire Queue component as it's only used for the database queue. If I use the Redis queue for example I don't need PDO. IMO ext-PDO should only be in the suggest section.

timacdonald replied:
I agree. I'm going through now and making tweaks from required to suggested 👍

Still a WIP.

timacdonald commented on src/Illuminate/Routing/composer.json:
```diff
@@ -15,7 +15,8 @@
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-filter": "*",
+        "ext-hash": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Routing/UrlGenerator.php#L333-L350

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Routing/UrlGenerator.php#L421-L436

timacdonald commented on src/Illuminate/Routing/composer.json:
```diff
@@ -15,7 +15,8 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-filter": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Routing/UrlGenerator.php#L623-L630

timacdonald commented on src/Illuminate/Session/composer.json:
```diff
@@ -15,7 +15,8 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-ctype": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Session/Store.php#L649-L652

timacdonald commented on src/Illuminate/Session/composer.json:
```diff
@@ -15,7 +15,8 @@
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-ctype": "*",
+        "ext-session": "*",
```
All session handlers extend `\SessionHandlerInterface`

timacdonald commented on src/Illuminate/Support/composer.json:
```diff
@@ -15,7 +15,8 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-ctype": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Support/Str.php#L1053-L1068

timacdonald commented on src/Illuminate/Support/composer.json:
```diff
@@ -15,7 +15,8 @@
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-ctype": "*",
+        "ext-filter": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Support/Stringable.php#L1175-L1178

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Support/Stringable.php#L260-L269

timacdonald commented on src/Illuminate/Testing/composer.json:
```diff
@@ -15,6 +15,7 @@
     ],
     "require": {
         "php": "^8.0.2",
+        "ext-mbstring": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Testing/Constraints/SeeInOrder.php#L41-L62

timacdonald commented on src/Illuminate/Validation/composer.json:
```diff
@@ -15,7 +15,8 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-filter": "*",
```
See usages of `filter_var`, `FILTER_FLAG_EMAIL_UNICODE`, `FILTER_FLAG_IPV4`, `FILTER_FLAG_IPV6`, `FILTER_VALIDATE_EMAIL`, `FILTER_VALIDATE_INT`, `FILTER_VALIDATE_IP`, and `FILTER_VALIDATE_MAC`.

timacdonald commented on src/Illuminate/Validation/composer.json:
```diff
@@ -15,7 +15,8 @@
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-filter": "*",
+        "ext-mbstring": "*",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Validation/Concerns/ValidatesAttributes.php#L2352-L2369

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Validation/DatabasePresenceVerifier.php#L103-L114

timacdonald commented on src/Illuminate/View/composer.json:
```diff
@@ -15,7 +15,7 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-tokenizer": "*",
```
See usages of `token_get_all`, `T_CLOSE_TAG`, `T_INLINE_HTML`, `T_OPEN_TAG`, and `T_OPEN_TAG_WITH_ECHO`.

timacdonald committed:
Moves ext-PDO to suggested for Queue

timacdonald committed:
Moves driver based extensions to suggested

timacdonald commented on src/Illuminate/Queue/composer.json:
```diff
@@ -39,6 +38,9 @@
     },
     "suggest": {
+        "ext-PDO": "Required to use the database queue worker.",
+        "ext-filter": "Required to use the SQS queue worker.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/SqsQueue.php#L205-L212

timacdonald commented on src/Illuminate/Queue/composer.json:
```diff
@@ -39,6 +38,9 @@
     "suggest": {
+        "ext-PDO": "Required to use the database queue worker.",
+        "ext-filter": "Required to use the SQS queue worker.",
+        "ext-mbstring": "Required to use the database failed job providers.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/Failed/DatabaseFailedJobProvider.php#L56-L65

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/Failed/DatabaseUuidFailedJobProvider.php#L56-L68

timacdonald commented on src/Illuminate/Queue/composer.json:
```diff
@@ -39,6 +38,9 @@
         }
     },
     "suggest": {
+        "ext-PDO": "Required to use the database queue worker.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/DatabaseQueue.php#L256-L280

// cc @X-Coder264

timacdonald commented on src/Illuminate/Filesystem/composer.json:
```diff
@@ -32,7 +32,9 @@
         }
     },
     "suggest": {
+        "ext-fileinfo": "Required to use the Filesystem class.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Filesystem/Filesystem.php#L450-L453

timacdonald commented on src/Illuminate/Filesystem/composer.json:
```diff
@@ -32,7 +32,9 @@
     "suggest": {
+        "ext-fileinfo": "Required to use the Filesystem class.",
         "ext-ftp": "Required to use the Flysystem FTP driver.",
+        "ext-hash": "Required to use the Filesystem class.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Filesystem/Filesystem.php#L175-L178

timacdonald commented on src/Illuminate/Database/composer.json:
```diff
@@ -36,6 +36,7 @@
         }
     },
     "suggest": {
+        "ext-filter": "Required to use the Postgres database driver.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Database/Query/Grammars/PostgresGrammar.php#L225-L251

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Database/Query/Grammars/PostgresGrammar.php#L666-L677

timacdonald commented on src/Illuminate/Console/composer.json:
```diff
@@ -35,6 +36,7 @@
         }
     },
     "suggest": {
+        "ext-pcntl": "Required to use signal trapping.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Signals.php#L77-L82

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Signals.php#L89-L102

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Console/Signals.php#L77-L82

timacdonald commented on src/Illuminate/Cache/composer.json:
```diff
@@ -34,6 +34,7 @@
         }
     },
     "suggest": {
+        "ext-filter": "Required to use the DynamoDb cache driver.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Cache/DynamoDbStore.php#L484-L495

timacdonald commented on src/Illuminate/Broadcasting/composer.json:
```diff
@@ -35,6 +34,7 @@
         }
     },
     "suggest": {
+        "ext-hash": "Required to use the Ably and Pusher broadcast drivers.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Broadcasting/Broadcasters/PusherBroadcaster.php#L43-L65

https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Broadcasting/Broadcasters/AblyBroadcaster.php#L107-L114

timacdonald committed:
Standardise capitalisation of ext-pdo

timacdonald commented on src/Illuminate/Database/composer.json:
```diff
@@ -16,7 +16,7 @@
     "require": {
         "php": "^8.0.2",
-        "ext-json": "*",
+        "ext-pdo": "*",
```
Search usages of `\PDO`, `\PDOStatement`, and `\PDOException`.

timacdonald commented on src/Illuminate/Queue/composer.json:
```diff
@@ -39,6 +38,9 @@
         }
     },
     "suggest": {
+        "ext-pdo": "Required to use the database queue worker.",
```
https://github.com/laravel/framework/blob/d3a4cd268bd716838482125f36ccf6bf8d51088a/src/Illuminate/Queue/DatabaseQueue.php#L256-L280

// cc @X-Coder264

timacdonald committed:
Update ext-pcntl suggest framework description

timacdonald committed:
Add additional framework required extensions

timacdonald committed:
Remove require-dev dependency on bcmath

timacdonald commented on composer.json:
```diff
@@ -83,6 +88,7 @@
         "illuminate/view": "self.version"
     },
     "require-dev": {
+        "ext-gmp": "*",
```
I think we still need this in `require-dev`.

See this PR: https://github.com/laravel/framework/pull/43959

Seems best to leave it as GMP objects have some quirks that this addresses and I can't replicate with standard classes.

timacdonald commented on composer.json:
```diff
@@ -16,8 +16,13 @@
     ],
     "require": {
         "php": "^8.0.2",
+        "ext-ctype": "*",
```
Required in `illuminate/support` and `illuminate\session`.

timacdonald commented on composer.json:
```diff
@@ -16,8 +16,13 @@
     "require": {
         "php": "^8.0.2",
+        "ext-ctype": "*",
+        "ext-filter": "*",
```
Required in:

- `illuminate/routing`
- `illuminate/validation`
- `illuminate/support`
- `illuminate/http`
- `illuminate/pagination`

Optional in:

- `illuminate/queue`
- `illuminate/cache`
- `illuminate/database`

timacdonald commented on composer.json:
```diff
@@ -16,8 +16,13 @@
         "php": "^8.0.2",
+        "ext-ctype": "*",
+        "ext-filter": "*",
+        "ext-hash": "*",
```
Required in:

- `illuminate/auth`
- `illuminate/routing`
- `illuminate/cookie`
- `illuminate/encryption`

Optional in:

- `illuminate/filesystem`
- `illuminate/broadcasting`

timacdonald commented on composer.json:
```diff
@@ -16,8 +16,13 @@
+        "ext-hash": "*",
         "ext-mbstring": "*",
         "ext-openssl": "*",
+        "ext-session": "*",
```
Required in:

- `illuminate/session`

timacdonald commented on composer.json:
```diff
@@ -16,8 +16,13 @@
         "ext-mbstring": "*",
         "ext-openssl": "*",
+        "ext-session": "*",
+        "ext-tokenizer": "*",
```
Required in:

- `illuminate/view`

timacdonald committed:
Adds ext-apcu to Cache suggests

timacdonald commented on src/Illuminate/Cache/composer.json:
```diff
@@ -34,6 +34,8 @@
         }
     },
     "suggest": {
+        "ext-apcu": "Required to use the APC cache driver.",
```
See the `ApcWrapper` class.

timacdonald committed:
code style
